core/drivers/gic.c

```diff
@@ -382,7 +382,7 @@ void gic_it_handle(struct gic_data *gd)
 
 	if (id == GIC_SPURIOUS_ID)
 		DMSG("ignoring spurious interrupt");
-	else
+	else if (id < gd->max_it)
```

Comment:
What about doing this instead:
```
if (id < gd->max_it)
    itr_handle(id)
else
    DMSG("ignoring interrupt %" PRIu32, id);
```

That way we'll see clearly if we're doing some unexpected spinning.